### ISSUE TITLE
cannot mount device

### ISSUE THREAD

**Issue by bksubhuti** [WoeUSB/WoeUSB#31]
## Describe the bug ##

i type this..
sudo /snap/bin/woe-usb.woeusb --device Win10.iso /dev/sda --tgt-fs NTFS
and get mount failure

i also get this without ntfs
## Expected behavior ##
i expect it to work
## Environment ##
Deepin 20
woe usb snap edge

bhante@subhuntu:~/Downloads$ sudo /snap/bin/woe-usb.woeusb --device Win10.iso /dev/sda --tgt-fs NTFS
WoeUSB v@@WOEUSB_VERSION@@
==============================
Mounting source filesystem...
mount: permission denied                                                                                      
mount_source_filesystem: Error: Unable to mount source media
Error: Unable to mount source filesystem                                                                      
Unmounting and removing "/media/woeusb_source_1611194231_17926"...                                            
umount: /media/woeusb_source_1611194231_17926: must be superuser to unmount                                   
cleanup_mountpoint: Warning: Unable to unmount "/media/woeusb_source_1611194231_17926".
Some mountpoints are not unmount/cleaned successfully and must be done manually                               
You may now safely detach the target device                                                                   
bhante@subhuntu:~/Downloads$

**Comment by JonnyTech** [comment]
The device `/dev/sda` is usually your boot drive so is already mounted, you do not want to destroy that! What is the output of`lsblk`?

**Comment by bksubhuti** [comment]
currently.. I have loop6 mounted.. for the iso and trying 3rd or 4th time to manually copy..so ignore that
sda is 8gb stick

bhante@subhuntu:~$ lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0         7:0    0  64.8M  1 loop /snap/gtk-common-themes/1514
loop1         7:1    0 162.9M  1 loop /snap/gnome-3-28-1804/145
loop2         7:2    0  97.9M  1 loop /snap/core/10583
loop3         7:3    0 109.1M  1 loop /snap/woe-usb/21
loop4         7:4    0   128M  1 loop /snap/signal-desktop/346
loop5         7:5    0  55.4M  1 loop /snap/core18/1944
loop6         7:6    0   5.8G  0 loop /home/bhante/win
sda           8:0    1   7.5G  0 disk 
└─sda1        8:1    1   7.5G  0 part /media/bhante/win10
nvme0n1     259:0    0 238.5G  0 disk 
├─nvme0n1p1 259:1    0   100M  0 part /boot/efi
├─nvme0n1p2 259:2    0    16M  0 part 
├─nvme0n1p3 259:3    0  78.1G  0 part /media/bhante/Acer
├─nvme0n1p4 259:4    0     1G  0 part 
├─nvme0n1p5 259:5    0    97G  0 part /
└─nvme0n1p7 259:6    0  62.2G  0 part /mnt/7914628E7BB468B5

**Comment by JonnyTech** [comment]
Already mounted, at `/media/bhante/win10`, do `sudo umount sda1`.

**Comment by bksubhuti** [comment]
I told you ,  i did the loop, but that was after the error.
In any case.. i know how to manually mount the win10.. so what is the command to make that work?
It is mounted in my home directory /home/bhante/win10
Please tell me the command to make this woeusb work.. i tried many combos.. this is driving me nuts.

Or how to do this manually?  Etcher says the iso does not have partition boot information.
I am not sure if this is intentional from microsoft so you are forced to use their creation tool.  But I downloaded from them just a few days ago.  

My windows is dead and it appears that many people have the same problem with the latest win update.

**Comment by JonnyTech** [comment]
- Unmount the iso
- Unmount the drive

`sudo ./woeusb --target-filesystem NTFS --device win10.iso /dev/sda`

**Comment by bksubhuti** [comment]
I removed the mount using gparted.
there was nothing mounted by the source, but it still gives error





bhante@subhuntu:~/Downloads$ sudo /snap/bin/woe-usb.woeusb --target-filesystem NTFS --device Win10.iso /dev/sda
WoeUSB v@@WOEUSB_VERSION@@
==============================
Mounting source filesystem...
mount: permission denied                                                                                      
mount_source_filesystem: Error: Unable to mount source media
Error: Unable to mount source filesystem                                                                      
Unmounting and removing "/media/woeusb_source_1611242072_21105"...                                            
umount: /media/woeusb_source_1611242072_21105: must be superuser to unmount                                   
cleanup_mountpoint: Warning: Unable to unmount "/media/woeusb_source_1611242072_21105".
Some mountpoints are not unmount/cleaned successfully and must be done manually                               
You may now safely detach the target device          


bhante@subhuntu:~/Downloads$ lsblk                                                                            
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
loop0         7:0    0  55.4M  1 loop /snap/core18/1944
loop1         7:1    0  97.9M  1 loop /snap/core/10583
loop2         7:2    0 162.9M  1 loop /snap/gnome-3-28-1804/145
loop3         7:3    0   128M  1 loop /snap/signal-desktop/346
loop4         7:4    0  64.8M  1 loop /snap/gtk-common-themes/1514
loop5         7:5    0 109.1M  1 loop /snap/woe-usb/21
sda           8:0    1   7.5G  0 disk 
nvme0n1     259:0    0 238.5G  0 disk 
├─nvme0n1p1 259:1    0   100M  0 part /boot/efi
├─nvme0n1p2 259:2    0    16M  0 part 
├─nvme0n1p3 259:3    0  78.1G  0 part /media/bhante/Acer
├─nvme0n1p4 259:4    0     1G  0 part 
├─nvme0n1p5 259:5    0    97G  0 part /
└─nvme0n1p7 259:6    0  62.2G  0 part /mnt/7914628E7BB468B5
bhante@subhuntu:~/Downloads$

**Comment by bksubhuti** [comment]
I tried also with sudo su.. but it also said i must be a super user.

**Comment by JonnyTech** [comment]
> mount: permission denied

Must be a snap issue. Try without snap:

```
wget https://github.com/WoeUSB/WoeUSB/releases/download/v5.0.4/woeusb-5.0.4.bash
chmod +x woeusb-5.0.4.bash
sudo umount -a
sudo ./woeusb-5.0.4.bash --target-filesystem NTFS --device win10.iso /dev/sda
```

**Comment by bksubhuti** [comment]
It failed ..  It got much further.  I remember I may have made a partition table as gpt from something I read.. I can try dos table and see if that helps.  Here is the output.

```
bhante@subhuntu:~$ sudo ./woeusb-5.0.4.bash --target-filesystem NTFS --device ~/Downloads/Win10.iso /dev/sda  
WoeUSB v5.0.4
==============================
Mounting source filesystem...
Wiping all existing partition table and filesystem signatures in /dev/sda...                                  
/dev/sda: 2 bytes were erased at offset 0x000001fe (dos): 55 aa                                               
/dev/sda: calling ioctl to re-read partition table: Success
Ensure that /dev/sda is really wiped...
Creating new partition table on /dev/sda...                                                                   
Creating target partition...                                                                                  
Making system realize that partition table has changed...                                                     
Wait 3 seconds for block device nodes to populate...                                                          
Cluster size has been automatically set to 4096 bytes.
Creating NTFS volume structures.
mkntfs completed successfully. Have a nice day.
--2021-01-22 11:29:31--  https://github.com/pbatard/rufus/raw/master/res/uefi/uefi-ntfs.img
Resolving github.com (github.com)... 52.74.223.119
Connecting to github.com (github.com)|52.74.223.119|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/pbatard/rufus/master/res/uefi/uefi-ntfs.img [following]
--2021-01-22 11:29:32--  https://raw.githubusercontent.com/pbatard/rufus/master/res/uefi/uefi-ntfs.img
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.8.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.8.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 524288 (512K) [application/octet-stream]
Saving to: ‘/tmp/WoeUSB.mAZpfF.tempdir/uefi-ntfs.img’

uefi-ntfs.img               100%[=========================================>] 512.00K  1.45MB/s    in 0.3s    

2021-01-22 11:29:33 (1.45 MB/s) - ‘/tmp/WoeUSB.mAZpfF.tempdir/uefi-ntfs.img’ saved [524288/524288]

1024+0 records in
1024+0 records out
524288 bytes (524 kB, 512 KiB) copied, 0.171072 s, 3.1 MB/s
Mounting target filesystem...
Copying files from source media...                                                                            
Installing GRUB bootloader for legacy PC booting support...                                                   
grub-install: error: /usr/lib/grub/i386-pc/modinfo.sh doesn't exist. Please specify --target or --directory.  
The command ""${command_grubinstall}" --target=i386-pc --boot-directory="${target_fs_mountpoint}" --force "${target_device}"" failed with exit status "1", program is prematurely aborted                                   
Unmounting and removing "/tmp/woeusb-source-20210122-112913-Friday.lzJW6B"...                                 
Unmounting and removing "/tmp/woeusb-target-20210122-112913-Friday.HdvDnl"...                                 
You may now safely detach the target device                                                                   
bhante@subhuntu:~$                           
```

**Comment by bksubhuti** [comment]
It booted and win install is working despite the fail message.  I will update this message after install success.

**Comment by JonnyTech** [comment]
> grub-install: error: /usr/lib/grub/i386-pc/modinfo.sh doesn't exist

You do not have grub installed correctly, but no worry as that is used for legacy boot. Thank you for confirming the success.

**Comment by bksubhuti** [comment]
If you want, i can help test for you.  It might be from a GPT partition table when i was trying many different things.
I can try again with MBR.  From what I remember, you cannot have a gpt and do legacy. 
Let me know if you want me to reformat the ptable and try again , or if you make changes and want me to try again.. you could probably test and reproduce this theory locally.

**Comment by JonnyTech** [comment]
No, the application wipes and repartitions the drive correctly. It is simply an issue with your Linux installation. I suspect that you did not follow the instructions to install the correct dependencies (https://github.com/WoeUSB/WoeUSB/wiki/Dependencies). Please close this issue.

**Comment by bksubhuti** [comment]
okay.. all is working fine then.  I appreciated your help.  Perhaps you can shut down the snap.

**Comment by JonnyTech** [comment]
Contact whoever distributes the snap and ask them to fix it, according to #3 it is @ErnyTech
